sdk/src/main/java/com/hashicorp/nomad/javasdk/NomadPredicates.java

```diff
@@ -112,4 +113,86 @@ public boolean apply(T value) {
             }
         };
     }
+
+    /**
+     * Returns a predicate that is true when either of the given predicates is true.
+     */
+    public static <T> Predicate<T> either(final Predicate<? super T> a, final Predicate<? super T> b) {
+        return new Predicate<T>() {
+            @Override
+            public boolean apply(T value) {
+                return a.apply(value) || b.apply(value);
+            }
+        };
+    }
+
+    /**
+     * Returns a predicate that checks if allocation has the given client status.
+     */
+    public static Predicate<AllocationListStub> allocationHasClientStatus(final String status) {
+        return new Predicate<AllocationListStub>() {
+            @Override
+            public boolean apply(@Nonnull AllocationListStub allocationListStub) {
+                return status.equals(allocationListStub.getClientStatus());
+            }
+        };
+    }
+
+    /**
+     * Returns a predicate that checks if allocation has completed.
+     */
+    public static Predicate<AllocationListStub> allocationHasCompleted() {
+        return allocationHasClientStatus("complete");
+    }
+
+    /**
+     * Returns a predicate that checks if allocation has failed.
+     */
+    public static Predicate<AllocationListStub> allocationHasFailed() {
+        return allocationHasClientStatus("failed");
+    }
+
+    /**
+     * Returns a predicate that checks if allocation has completed successfully or failed (i.e. is not in progress).
+     */
+    public static Predicate<AllocationListStub> allocationFinishedRunning() {
+        return either(allocationHasCompleted(), allocationHasFailed());
```

Comment:
I think it would be nice to have `allocationFinishedRunning` as a predicate. But all your recommendations would definitely be nice.